src/shared/services/api-key.service.ts

```diff
@@ -0,0 +1,62 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { Config } from 'src/config/config';
+import { HistoryFilter } from 'src/payment/models/history/dto/history-filter.dto';
+import { HistoryQuery } from 'src/payment/models/history/dto/history-query.dto';
+import { User } from 'src/user/models/user/user.entity';
+import { Util } from '../util';
+
+@Injectable()
+export class ApiKeyService {
+  constructor() {}
+
+  public createKey(address: string, queryFilter: HistoryFilter): string {
+    let hash = Util.createHash(Util.createHash(address + new Date().toISOString(), 'sha256'), 'md5').toUpperCase();
+
+    return hash.substring(0, hash.length - 5) + this.getFilterHash(queryFilter) + Config.apiKeyVersionCT;
```

Comment:
We should try to store this magic numbers (in this case 5) into variables.

Reply:
You mean as a variable in config?